### PR TITLE
Fix ATS rounding

### PR DESCRIPTION
**Pull request by mlsteele**
An attempt to send $2.32 would respond that you only had $2.32 available to send. Fix this by fixing truncation in FmtAmount. This PR is a vendor and regression test. The fix is in stellarnet.

With this fix it may still be possible to get a similar nonsensical response. But due to the precision involved, it is significantly less likely. I'm inclined to not do work to dredge up that case until it is seen.

### CODE REVIEW

File: go/vendor/github.com/keybase/stellarnet/format.go [hunk 1]
```diff
@@ -54,19 +54,21 @@ func FmtAmount(amount string, precisionTwo bool, rounding FmtRoundingBehavior) (
...
 		precision = 2
 	}
 	var s string
 	if rounding == Round {
-		s = x.FloatString(precision)
+		s = x.StringFixed(precision)
 	} else {
-		s = x.FloatString(precision + 1)
+		// bigdecimal is used instead of big.Rat because big.Rat.FloatString rounds the last digit.
+		// Which would turn 2.9999999 into 3.0 when we want 2.9.
```
Comment (xgess) [inline go/vendor/github.com/keybase/stellarnet/format.go]: is this an extreme example, or is it always this aggressive in chopping off pennies?

File: go/stellar/stellarsvc/frontend_test.go [hunk 2]
```diff
@@ -1866,6 +1866,50 @@ func TestBuildPaymentLocal(t *testing.T) {
...
+		From:     senderAccountID,
+		To:       "t_alice",
+		Amount:   "2.32",
+		Currency: &usd,
+	})
+	require.NoError(t, err)
+	t.Logf(spew.Sdump(bres))
+	require.Equal(t, false, bres.ReadyToReview)
+	require.Equal(t, "", bres.ToErrMsg)
+	// Before the fix AmountErrMsg had $2.32
```
Comment (xgess) [inline go/stellar/stellarsvc/frontend_test.go]:
```suggestion
	// Before the fix, AmountErrMsg had $2.32
```

File: go/vendor/github.com/keybase/stellarnet/format.go (hunk 1 above)
Comment (mlsteele) [reply go/vendor/github.com/keybase/stellarnet/format.go]: Both the example you deleted and the one you replaced it with are accurate, eh?
https://goplay.space/#qqTM1nZco_z

Comment (mlsteele) [reply go/vendor/github.com/keybase/stellarnet/format.go]: The issue with FloatString was revealed to me in two stages.
First, the second least significant digit out of FloatString can change due to rounding: `(2.1299).FloatString(3)[:-1] -> 2.13`.
Second, all the digits out of FloatString can change due to rounding: `(2.9999).FloatString(3)[:-1] -> 3.00`

Comment (xgess) [reply go/vendor/github.com/keybase/stellarnet/format.go]: huh. ok.

**Pull request merged**